detectron2/data/transforms/transform_gen.py

```diff
@@ -113,6 +114,33 @@ def __repr__(self):
     __str__ = __repr__
 
 
+class RandomApply(TransformGen):
+    """
+    Randomly apply the wrapper transformation with a given probability.
+    """
+
+    def __init__(self, transform, prob=0.5):
+        """
+        Args:
+            prob (float): probability between 0.0 and 1.0 that
```

Comment:
document what `transform` can be and check its type in `__init__`

Reply:
done